x-pack/platform/plugins/shared/index_management/public/application/hooks/use_convert_index_to_lookup/use_convert_index_to_lookup.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState, useCallback, useRef } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { ReindexStatus } from '@kbn/upgrade-assistant-pkg-common';
+
+import { getReindexStatus, startReindex } from '../../services';
+
+const POLL_INTERVAL = 3000;
+
+interface UseConvertIndexToLookupArgs {
+  sourceIndexName: string;
+  onSuccess: (newIndexName: string) => void;
+  onClose: () => void;
+}
+
+export const useConvertIndexToLookup = ({
+  sourceIndexName,
+  onSuccess,
+  onClose,
+}: UseConvertIndexToLookupArgs) => {
+  const [isConverting, setIsConverting] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const onSuccessRef = useRef(onSuccess);
+  const onCloseRef = useRef(onClose);
+
+  useEffect(() => {
+    onSuccessRef.current = onSuccess;
+    onCloseRef.current = onClose;
+  }, [onSuccess, onClose]);
+
+  const pollIntervalIdRef = useRef<number | null>(null);
+
+  const clearPollInterval = useCallback(() => {
+    if (pollIntervalIdRef.current) {
+      clearTimeout(pollIntervalIdRef.current);
+      pollIntervalIdRef.current = null;
+    }
+  }, []);
+
+  const updateStatus = useCallback(async () => {
+    clearPollInterval();
+
+    const { data, error } = await getReindexStatus(sourceIndexName);
+
+    if (error) {
+      setErrorMessage(error.message);
+      setIsConverting(false);
+      return;
+    }
+
+    if (data?.reindexOp?.status === ReindexStatus.inProgress) {
+      pollIntervalIdRef.current = window.setTimeout(updateStatus, POLL_INTERVAL);
+    } else if (
+      data?.reindexOp?.status === ReindexStatus.failed ||
+      data?.reindexOp?.status === ReindexStatus.cancelled
+    ) {
+      setErrorMessage(
+        data?.reindexOp?.errorMessage ??
+          i18n.translate(
+            'xpack.idxMgmt.convertToLookupIndexAction.reindexFailedOrCancelledErrorMessage',
+            {
+              defaultMessage: 'Reindex did not complete successfully.',
```

Comment:
Might be good to say `Reindex did not complete successfully due to unknown error.` since we do return the message if its provided